### ISSUE TITLE
Unused import triggers no-unused-vars rule

### ISSUE THREAD

**Issue by somebody1234** [denoland/deno_lint#730]
### Lint Name

no-unused-vars

### Code Snippet

```ts
import { anything } from './file.ts';
```

### Expected Result
Warning related to unused import

### Actual Result
no-unused-vars warning

### Additional Info
To be specific, renaming... certainly works, but isn't as simple as the warning suggests

### Version
deno 1.11.0 (release, x86_64-unknown-linux-gnu)
v8 9.1.269.35
typescript 4.3.2

**Comment by magurotuna** [comment]
This behavior is in accordance with [ESLint's `no-unused-vars` rule](https://eslint.org/demo#eyJ0ZXh0IjoiaW1wb3J0IGZvbyBmcm9tIFwiYmFyXCI7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJjb25zdHJ1Y3Rvci1zdXBlciI6MiwiZm9yLWRpcmVjdGlvbiI6MiwiZ2V0dGVyLXJldHVybiI6Miwibm8tYXN5bmMtcHJvbWlzZS1leGVjdXRvciI6Miwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOjIsIm5vLWNsYXNzLWFzc2lnbiI6Miwibm8tY29tcGFyZS1uZWctemVybyI6Miwibm8tY29uZC1hc3NpZ24iOjIsIm5vLWNvbnN0LWFzc2lnbiI6Miwibm8tY29uc3RhbnQtY29uZGl0aW9uIjoyLCJuby1jb250cm9sLXJlZ2V4IjoyLCJuby1kZWJ1Z2dlciI6Miwibm8tZGVsZXRlLXZhciI6Miwibm8tZHVwZS1hcmdzIjoyLCJuby1kdXBlLWNsYXNzLW1lbWJlcnMiOjIsIm5vLWR1cGUtZWxzZS1pZiI6Miwibm8tZHVwZS1rZXlzIjoyLCJuby1kdXBsaWNhdGUtY2FzZSI6Miwibm8tZW1wdHkiOjIsIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tZW1wdHktcGF0dGVybiI6Miwibm8tZXgtYXNzaWduIjoyLCJuby1leHRyYS1ib29sZWFuLWNhc3QiOjIsIm5vLWV4dHJhLXNlbWkiOjIsIm5vLWZhbGx0aHJvdWdoIjoyLCJuby1mdW5jLWFzc2lnbiI6Miwibm8tZ2xvYmFsLWFzc2lnbiI6Miwibm8taW1wb3J0LWFzc2lnbiI6Miwibm8taW5uZXItZGVjbGFyYXRpb25zIjoyLCJuby1pbnZhbGlkLXJlZ2V4cCI6Miwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOjIsIm5vLW1pc2xlYWRpbmctY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1taXhlZC1zcGFjZXMtYW5kLXRhYnMiOjIsIm5vLW5ldy1zeW1ib2wiOjIsIm5vLW9iai1jYWxscyI6Miwibm8tb2N0YWwiOjIsIm5vLXByb3RvdHlwZS1idWlsdGlucyI6Miwibm8tcmVkZWNsYXJlIjoyLCJuby1yZWdleC1zcGFjZXMiOjIsIm5vLXNlbGYtYXNzaWduIjoyLCJuby1zZXR0ZXItcmV0dXJuIjoyLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6Miwibm8tc3BhcnNlLWFycmF5cyI6Miwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOjIsIm5vLXVuZGVmIjoyLCJuby11bmV4cGVjdGVkLW11bHRpbGluZSI6Miwibm8tdW5yZWFjaGFibGUiOjIsIm5vLXVuc2FmZS1maW5hbGx5IjoyLCJuby11bnNhZmUtbmVnYXRpb24iOjIsIm5vLXVudXNlZC1sYWJlbHMiOjIsIm5vLXVudXNlZC12YXJzIjoyLCJuby11c2VsZXNzLWNhdGNoIjoyLCJuby11c2VsZXNzLWVzY2FwZSI6Miwibm8td2l0aCI6MiwicmVxdWlyZS15aWVsZCI6MiwidXNlLWlzbmFuIjoyLCJ2YWxpZC10eXBlb2YiOjJ9LCJlbnYiOnt9fX0=).
Indeed it makes more sense to rename it to `no-unused-imports`, but in my opinion it doesn't really matter if we leave the rule name as-is, because `import` statements effectively introduce read-only "variable" from external modules.

**Comment by somebody1234** [comment]
@magurotuna i don't mind that; it's still important to note that the suggested fix is wrong

**Comment by somebody1234** [comment]
could possibly also see if eslint could split that lint into `no-unused-imports`

**Comment by magurotuna** [comment]
@somebody1234 
Could you provide a specific example please? I don't understand what you mean by "the fix for them is different".
Perhaps you mean that the hint should be fixed? i.e. the current hint `` If this is intentional, prefix it with an underscore like `_anything` `` could be better, like ``If this is intentional, consider aliasing it like `anything as _anything`  ``?

> same problem, although a bit less of a problem, with methods - especially if you have an empty default implementation, which will show up in the docs, so you will not want a preceding underscore

Again I can't imagine a specific scenario, but it sounds like you should use `deno-lint-ignore` directive in that place to suppress the linter, because you have a clear reason to do so and that's a little too special case to be covered by the linter.

**Comment by somebody1234** [comment]
yes, by "fix" i mean the fix mentioned in the hint.  
and i really doubt it's "a little too special to be covered by the linter" - especially since the linter can probably already find the location where a variable is defined (and therefore whether it's an import, a parameter, or a local variable)

**Comment by magurotuna** [comment]
> yes, by "fix" i mean the fix mentioned in the hint.

OK, that makes sense to me too. I'll make the hint clear.

> and i really doubt it's "a little too special to be covered by the linter" - especially since the linter can probably already find the location where a variable is defined (and therefore whether it's an import, a parameter, or a local variable)

I don't really get the point... how does the linter's capability of finding variable declaration location have something to do with your previous explanation about methods with empty default implementation? Could you show me an example?

**Comment by somebody1234** [comment]
well my point is, it shouldn't be *too* hard to make this change in the linter

**Comment by magurotuna** [comment]
Which change? I don't ever get your point as your explanation sounds too abstract to me.

Could I have the following resources?

- sample code snippet including "an empty default implementation"
- what you expect the linter to do for the snippet
- the current behavior of the linter for the snippet

**Comment by magurotuna** [comment]
@somebody1234 I have fixed the issue regarding the hint for named imports and opened #734. Do you think this PR addresses all issues you're having?

**Comment by somebody1234** [comment]
yea, looks great, thanks
